packages/x-components/src/x-modules/facets/service/facets.service.ts

```diff
@@ -83,23 +85,19 @@ export class DefaultFacetsService implements FacetsService {
    * to the selected state.
    *
    * @param facetsGroup - The {@link FacetsGroup} to set into the store state.
-   * @returns Object with the `previousFilters` removed from store state and the `newFilters` to
-   * set into the store state.
+   * @returns An array with the new filters.
    * @internal
    */
-  protected updateStore(facetsGroup: FacetsGroup): {
-    previousFilters: Filter[];
-    newFilters: Filter[];
-  } {
+  protected updateStore(facetsGroup: FacetsGroup): Filter[] {
     this.removeGroupFacets(facetsGroup.id);
-    const previousFilters = this.removeGroupFilters(facetsGroup.id);
+    this.removeGroupFilters(facetsGroup.id);
     facetsGroup.facets.forEach(facet => {
       this.setFacetGroup({ facetId: facet.id, groupId: facetsGroup.id });
       this.setFacet(facet);
     });
     const newFilters = facetsGroup.facets.flatMap(facet => facet.filters);
     this.setFilters(newFilters);
-    return { newFilters, previousFilters };
+    return newFilters;
```

Comment:
I don't remember why we needed the `previousFilters` for.

Reply:
We used them to keep the previous selected state of the filters in the `updateFacets` method, but then this method was changed to first retrieve the selected filters, then replacing the filters in the store, and finally updating the filters selected state. I don't know the reason of this change, but I prefer it, as it avoids us to return 2 things from the same method. I don't think there is any issue with this approach.